code/game/objects/structures/doors/_door.dm

```diff
@@ -41,26 +36,23 @@
 	. = 10
 
 /obj/structure/door/on_update_icon()
-	..()
-	if(density)
-		icon_state = "[icon_base]"
-	else
-		icon_state = "[icon_base]open"
+	. = ..()
+	icon_state = "[icon_base][!density ? "open" : null]"
```

Comment:
"", not null. Why `. = ..()`?

Reply:
Afaik most on_update_icons() use `. = ..()`, so I stick to that pattern.

Reply:
From reading @MistakeNot4892's structure icon update PR this is not true, and as far as I'm aware there is no meaningful return value in this proc.

Reply:
This is correct, on_update_icon() does not usefully return.